pact/Pact/Core/Errors.hs

```diff
@@ -1061,6 +1095,586 @@ deriveConstrInfo ''EvalError
+  InvariantArgLengthMismatch fqn expected got ->
+    thsep
+      ["Argument length mismatch for", renderFullyQualName fqn, ", expected"
+      , tInt expected <> ","
+      , "got"
+      , tInt got]
+  InvariantInvalidManagedCapKind msg ->
+    thsep ["Invalid managed cap kind", msg]
+  InvariantNoSuchKeyInTable tbl (RowKey rk) ->
+    thsep
+      ["No such key"
+      , rk
+      , "in table"
+      , renderTableName tbl]
+  InvariantEmptyCapStackFailure ->
+    "Attempted to pop or manipulate the capstack, but it was found to be empty"
+
+hyperlaneErrorToBoundedText' :: HyperlaneError -> Text
+hyperlaneErrorToBoundedText' = \case
+  HyperlaneErrorFailedToFindKey key ->
+    thsep ["Failed to find key in object:", _field key]
+  HyperlaneErrorNumberOutOfBounds key ->
+    thsep ["Object key", _field key, "was out of bounds"]
+  HyperlaneErrorBadHexPrefix key ->
+    thsep ["Missing 0x prefix on field", _field key]
+  HyperlaneErrorInvalidBase64 key ->
+    thsep ["Invalid base64 encoding on field", _field key]
+  HyperlaneErrorIncorrectSize key expected actual ->
+    thsep
+      ["Incorrect binary data size"
+      , _field key <> ". Expected:"
+      , tInt expected <> ", but got"
+      , tInt actual]
+  -- Library decoding error: do not serialise
+  HyperlaneErrorInvalidChainId _ ->
+    thsep ["Failed to decode chainId"]
+
+hyperlaneDecodeErrorToBoundedText' :: HyperlaneDecodeError -> Text
+hyperlaneDecodeErrorToBoundedText' = \case
+  HyperlaneDecodeErrorBase64 -> "Failed to base64-decode token message"
+  -- Library error, do not serialise
+  HyperlaneDecodeErrorInternal _ ->
+    "Hyperlane Native Internal Decoding error"
+  HyperlaneDecodeErrorBinary -> "Decoding error: binary decoding failed"
+  HyperlaneDecodeErrorParseRecipient -> "Could not parse recipient into a guard"
+
+-- | NOTE: Do _not_ change this function post mainnet release just to improve an error.
+--  This will fork the chain, these messages will make it into outputs.
+--  See [Bounded Text Errors]
+evalErrorToBoundedText :: EvalError -> BoundedText PactErrorMsgSize
+evalErrorToBoundedText = mkBoundedText . \case
+  ArrayOutOfBoundsException len ix ->
+    thsep
+    [ "Array index out of bounds. Length"
+    , tInt len <> ","
+    , "Index"
+    , tInt ix]
+  ArithmeticException txt ->
+    thsep ["Arithmetic exception:", txt]
+  EnumerationError txt ->
+    thsep ["Enumeration error:", txt]
+  DecodeError txt ->
+    thsep ["Decoding error:", txt]
+  FloatingPointError txt ->
+    thsep ["Floating point error:", txt]
+  -- Todo: probably enhance this data type
+  GasExceeded ->
+    "Gas limit exceeded"
+  InvariantFailure msg ->
+    thsep
+      ["Execution Invariant error, please report this to the pact team @kadena:", invariantErrorToBoundedText' msg]
+  NativeArgumentsError (NativeName n) tys ->
+    thsep (["Native evaluation error for native", n <> ",", "received incorrect argument(s) of type(s)"] ++ (renderArgTypeError <$> tys))
```

Comment:
How about this phrasing:
"Type error: failed to call native function `somenative` with argument(s) with types(s): ..."